hypervisor/hypervisor.go

```diff
@@ -20,8 +20,14 @@ func (ctx *VmContext) loop() {
 			glog.V(1).Info("got nil event.")
 			continue
 		}
+		ctx.lock.Lock()
+		if ctx.handler == nil {
+			ctx.lock.Unlock()
```

Comment:
why not use a defer to simplify the code?

Reply:
@allencloud I only want to lock `ctx.handler(ctx, ev)` here, if we use lock and defer, the whole event loop will be in the lock, especially waiting an event from `ev, ok := <-ctx.Hub`.

BTW, I'm not sure this patch is still on the correct direction. /cc @laijs